### PR TITLE
[Extension] Add CatalogType to the list of functions generated in `extension_entries.hpp`

### PR DESCRIPTION
**Pull request by Tishj**
This PR fixes https://github.com/duckdblabs/duckdb-internal/issues/155

This should improve the error message that users gets when they are misusing a function that lives in an extension.
For example, the provided test:
```
statement error
FROM json("['item':'phasers','year':2155','count':1035]");
----
Catalog Error: Table Function with name "json" is not in the catalog, a function by this name exists in the json extension, but it's of a different type, namely Macro Function
```

I also took the opportunity to clean up the `scripts/generate_extensions_function.py` that we use to generate this file
It should be much more trivial and straightforward to extend this script to the other remaining catalog entries of extensions.

### CODE REVIEW

File: src/include/duckdb/main/extension_entries.hpp [hunk 1]
```diff
@@ -20,208 +20,209 @@ struct ExtensionEntry {
...
+    {"tpcds_answers", "tpcds", "table"},
+    {"tpcds_queries", "tpcds", "table"},
+    {"tpch", "tpch", "pragma"},
+    {"tpch_answers", "tpch", "table"},
+    {"tpch_queries", "tpch", "table"},
+}; // END_OF_EXTENSION_FUNCTIONS
 
 static constexpr ExtensionEntry EXTENSION_SETTINGS[] = {
     {"azure_storage_connection_string", "azure"},
-    {"binary_as_string", "parquet"},
```
Comment (Tishj) [inline src/include/duckdb/main/extension_entries.hpp:210]: Not sure what happened here?
I built with `DISABLE_BUILTIN_EXTENSIONS=1 BUILD_ALL_EXT=1 make release` as I have also added to the script as an error message when the prerequisites arent met

File: src/include/duckdb/main/extension_helper.hpp [hunk 2]
```diff
@@ -87,6 +87,21 @@ class ExtensionHelper {
 	static string GetExtensionName(const string &extension);
 	static bool IsFullPath(const string &extension);
 
+	//! Lookup a name + type in an ExtensionFunctionEntry list
+	template <size_t N>
+	static pair<string, string> FindExtensionInFunctionEntries(const string &name,
```
Comment (Tishj) [inline src/include/duckdb/main/extension_helper.hpp:92]: Hmm this actually doesn't work correctly, I noticed `json_execute_serialized_sql` exists both as a `pragma` and as a `table` function.
This should probably return a vector and we need to check if none of the entries match the requested function type.

File: src/include/duckdb/main/extension_entries.hpp (hunk 1 above)
Comment (Tishj) [reply src/include/duckdb/main/extension_entries.hpp:210]: I have manually added the missing setting from the parquet extension for now, we should probably update the notes in the generation script about this though

File: src/include/duckdb/main/extension_entries.hpp [hunk 3]
```diff
@@ -20,208 +20,217 @@ struct ExtensionEntry {
...
-                                                         {"tpcds", "tpcds"},
-                                                         {"tpcds_answers", "tpcds"},
-                                                         {"tpcds_queries", "tpcds"},
-                                                         {"tpch", "tpch"},
-                                                         {"tpch_answers", "tpch"},
-                                                         {"tpch_queries", "tpch"}}; // END_OF_EXTENSION_FUNCTIONS
+struct ExtensionFunctionEntry {
+	char name[48];
+	char extension[48];
+	char type[48];
```
Comment (Mytherin) [inline src/include/duckdb/main/extension_entries.hpp:26]: Can we make this an enum instead of a string?

**Comment by Tishj** [comment]
@Mytherin the regression test failure seems unrelated, not part of code touched by this PR, could you re-run it or can we ignore it?

It seems that the extension entries jobs still ran

**Comment by Mytherin** [comment]
Thanks! LGTM - good idea to keep track of this

**Pull request merged**